common/Latin.yml

```
@@ -6,5 +6,7 @@ nonword: true
 action:
   name: replace
 swap:
-  '\b(?:eg|e\.g\.)[\s,]': for example
-  '\b(?:ie|i\.e\.)[\s,]': that is
+  '\b(?:eg|e\.g\.)[\s]': 'for example '
+  '\b(?:eg|e\.g\.)[,]': 'for example'
+  '\b(?:ie|i\.e\.)[\s]': 'that is '
+  '\b(?:ie|i\.e\.)[,]': 'that is'
```

there is a problem according to me

It should be either

```suggestion
  '\b(?:ie|i\.e\.),?': 'that is'
```

I don't see the point of catching a square backet made of only one item

---

My problem here is that you went from

```
  '\b(?:eg|e\.g\.)[\s,]': for example
  '\b(?:ie|i\.e\.)[\s,]': that is
```

and you should have used

```
 '\b(?:eg|e\.g\.)\s*,?\s*': "for example "
 '\b(?:ie|i\.e\.)\s*,?\s*': "that is "
```

---

I think I didn't like leaving the comma outside the bracket on its own, I can remove it.

However, the question can't be there because it conflicts with the rule above. Then the fix would be useless.
It would once again correct ``i.e. daffodils`` to ``that isdaffodils``

---

it will catch `i.e. `, sure,, but look, it will add a space at the end

---

but once again, don't reinvent the wheel, look at official packages and copy paste

https://github.com/errata-ai/Google/blob/28adc48796277429c7d0cc80932569c5e7b0c360/Google/Latin.yml#L4

---

``i.e. `` is already caught by the rule above it though? The rule above corrects it with proper spacing.

---

well, the more I look at the Google's regexp, the more I'm doubtful

there is one thing interesting to use anyway, it's the fact there could be a coma or a semicolon, it's very frequent with Latin expression

https://github.com/search?q=language%3AMarkdown+%22+i.e.%3B%22&type=code

---

Is there a particular reason why you write the swap rules like:
``'a': "b"``
Mine generally are: 
``'a': 'b'``

---

@ccoVeille, I'm hesitant with your suggestion. While it reduces the rules from 4 to 2, it also points to errors where it shouldn't since anything after ``e.g`` is now a problem, such as ``e.g.z``. I realize that such an expression is hardly acceptable, but this is a rule where the level is "error" and therefore requires the upmost precision.

@tbazant, I tried to maintain the rule as it is by limiting it only to spaces and commas, but I may have been over-zealous in doing so. Can we expand it to include practically any character that may come after ``e.g`` or ``i.e.``? I'd rather act with your blessing.

---

> Is there a particular reason why you write the swap rules like:
> ``'a': "b"``
> Mine generally are: 
> ``'a': 'b'``

They are equivalent, no real reason to use one or another

---

> @ccoVeille, I'm hesitant with your suggestion. While it reduces the rules from 4 to 2, it also points to errors where it shouldn't since anything after ``e.g`` is now a problem, such as ``e.g.z``. I realize that such an expression is hardly acceptable, but this is a rule where the level is "error" and therefore requires the upmost precision.

That's right. I think you should consider maintaining files used for tests in your repository

---

Please try this and let me know

```yaml
'\b(?:ie|i\.e\.)\s*[,;]?\s+': 'that is '
```

I think it should catch the examples you spotted.

---

the last one had issue

The more I try, the uglier the regexp becomes

```yaml
 '\b(?:ie[,;]\s*|i\.e\.\s*[,;\s]+)\s+': 'that is '
 '\b(?:eg[,;]\s*|e\.g\.\s*[,;\s]+)\s+': 'for example '
```

It's important to focus on a test document, that will reference what you expect to catch
```
Whatever, i.e., foo
Whatever, i.e.; foo
Whatever, i.e. foo

Whatever, ie., foo
Whatever, ie.; foo
Whatever, ie. foo

Whatever, ie, foo
Whatever, ie; foo
Whatever, ie foo
Whatever, ie foo


Nothing should match after this line:
Whatever, ie.z foo
Ireland TLD is .ie
Whatever, ie
Internet Explorer's nickname was IE.
```

You approach to split on multiple line was good